components/ingest-service/migration/migration.go

```diff
@@ -270,17 +289,25 @@ func (ms *Status) update(s string) {
 }
 
 // update the migration status message with an error message (private)
-func (ms *Status) updateErr(err, s string) {
-	logFatal(err, s)
+func (ms *Status) updateErr(errMessage, s string) {
+	logFatal(errMessage, s)
 	ms.status = "Error: " + s
 	ms.finished = true
+	err := ms.unlock()
+	if err != nil {
+		logFatal(err.Error(), "Failed to unlock migration in postgresql")
+	}
 }
 
 // finish updates the status and marks the migration as finished
 func (ms *Status) finish(s string) {
 	logInfo(s)
 	ms.status = s
 	ms.finished = true
+	err := ms.unlock()
+	if err != nil {
+		logFatal(err.Error(), "Failed to unlock migration in postgresql")
+	}
```

Comment:
The migration lock is removed when the finish is called to prevent background migration from running on two ingest services.